pytest/tests/sanity/replay_chain_from_archive.py

```diff
@@ -0,0 +1,143 @@
+
+EPOCH_LENGTH = 10
+GC_NUM_EPOCHS_TO_KEEP = 3
+GC_BLOCKS_LIMIT = 5
+
+# Shard layout with 5 roughly equal size shards for convenience.
+SHARD_LAYOUT = {
+    "V1": {
+        "boundary_accounts": [
+            "fff",
+            "kkk",
+            "ppp",
+            "uuu",
+        ],
+        "version": 2,
+        "shards_split_map": [],
+        "to_parent_shard_map": [],
+    }
+}
+
+NUM_SHARDS = len(SHARD_LAYOUT["V1"]["boundary_accounts"]) + 1
+
+ALL_ACCOUNTS_PREFIXES = [
+    "aaa",
+    "ggg",
+    "lll",
+    "rrr",
+    "vvv",
+]
+
+
+class ReplayChainFromArchiveTest(unittest.TestCase):
+
+    def setUp(self):
+        node_config_dump, node_config_sync = state_sync_lib.get_state_sync_configs_pair(
+        )
+
+        # Validator node configs: Enable single-shard tracking with memtries enabled.
+        node_config_sync["tracked_shards"] = []
+        configs = {x: node_config_sync for x in range(4)}
+
+        # Dump+RPC node config: Enable tracking all shards with memtries enabled.
+        node_config_dump["tracked_shards"] = [0]
+        configs[4] = node_config_dump
+
+        # Archival node config: Enable tracking all shards with memtries enabled.
+        node_config_archival = copy.deepcopy(node_config_dump)
+        node_config_archival["archive"] = True
+        configs[5] = node_config_archival
+
+        # Configure GC to make sure it runs before the random workload finishes.
+        for i in configs.keys():
+            configs[i]["gc_num_epochs_to_keep"] = GC_NUM_EPOCHS_TO_KEEP
+            configs[i]["gc_blocks_limit"] = GC_BLOCKS_LIMIT
+            configs[i]["gc_step_period"] = {"secs": 0, "nanos": 100000000}
+
+        self.nodes = cluster.start_cluster(
+            num_nodes=4,
+            num_observers=2,
+            num_shards=NUM_SHARDS,
+            config=None,
+            genesis_config_changes=[["genesis_height", 0],
+                                    ["epoch_length", EPOCH_LENGTH],
+                                    ["shard_layout", SHARD_LAYOUT],
+                                    ["block_producer_kickout_threshold", 0],
+                                    ["chunk_producer_kickout_threshold", 0]],
+            client_config_changes=configs)
+        self.assertEqual(6, len(self.nodes))
+
+        self.rpc_node = self.nodes[4]
+        self.archival_node = self.nodes[5]
+
+    def test(self):
+        self.testcase = simple_test.SimpleTransferBetweenAccounts(
+            nodes=self.nodes,
+            rpc_node=self.rpc_node,
+            account_prefixes=ALL_ACCOUNTS_PREFIXES,
+            epoch_length=EPOCH_LENGTH)
+
+        self.testcase.wait_for_blocks(5)
+
+        self.testcase.create_accounts()
+
+        self.testcase.deploy_contracts()
+
+        # Run the test until there are enough number of epochs to trigger garbage collection.
+        self.testcase.random_workload_until(EPOCH_LENGTH *
+                                            (GC_NUM_EPOCHS_TO_KEEP + 2) + 5)
```

Comment:
Does it make sense to replace `5`  with `GC_BLOCKS_LIMIT`?

Reply:
done.